pkg/commands/artifact/run.go

```diff
@@ -609,6 +610,12 @@ func Exit(opts flag.Options, failedResults bool) {
 	}
 }
 
+func ExitOnEosl(opts flag.Options, m types.Metadata) {
+	if opts.ReportOptions.ExitOnEosl && opts.ExitCode != 0 && m.OS != nil && m.OS.Eosl {
+		os.Exit(opts.ExitCode)
+	}
```

Comment:
It might be better to use the existing `Exit`.

```suggestion
	if opts.ReportOptions.ExitOnEosl && m.OS != nil && m.OS.Eosl {
		Exit(opts, true)
	}
```